Diplom3

TimurSharibov opened the pull request.

TimurSharibov committed:
add tests

TimurSharibov committed:
add RegistrationTest

TimurSharibov committed:
add pages

TimurSharibov committed:
add LoginTests

TimurSharibov committed:
add OpenProfileTests

TimurSharibov committed:
add ConstructorTests

TimurSharibov committed:
add deleteuser

TimurSharibov committed:
commit before improving the code structure
All tests run

TimurSharibov committed:
Changed structure and added @Step annotation
All tests runs

TimurSharibov committed:
Changed structure and added @Step annotation
All tests runs - 2

TimurSharibov committed:
add-allure report

TimurSharibov committed:
далены локальные файлы, обновлен .gitignore, добавлен Readme.md

TimurSharibov committed:
add README.md

nikolay-977 commented on src/test/java/test/ConstructorTest.java:
```diff
@@ -0,0 +1,73 @@
+    }
+
+    @Test
+    @Step("Переход во вкладку Начинки")
```
Необходимо исправить: При реализации тестов используй аннотацию @DisplayName("Авторизация существующего пользователя"). В скобках указывай название теста. Это сделает отчет Allure информативнее, что позволит сократить время локализации дефекта. Аннотацию `@Step` следует использовать при реализации шагов.

nikolay-977 commented on src/test/java/test/LoginTest.java:
```diff
@@ -0,0 +1,124 @@
+import static java.time.Duration.ofSeconds;
+import static org.junit.Assert.assertTrue;
+
+public class LoginTest {
```
Можно улучшить: применить наследование - создать класс BaseTest и вынести в него методы с аннотациями `@Before` и `@After`, где это возможно. Это позволит избежать повторяющегося кода и сократит код автотестов.

nikolay-977 commented on src/test/java/test/RegistrationTest.java:
```diff
@@ -0,0 +1,116 @@
+        registrationPage.enterPassword(password);
+        registrationPage.clickRegistrationButton();
+
+        new WebDriverWait(webDriver, ofSeconds(10)).until(ExpectedConditions.urlToBe("https://stellarburgers.nomoreparties.site/login"));
```
Можно улучшить: использовать Implicit wait, задается один раз для всего проекта:
driver.manage().timeouts().implicitlyWait(Duration.ofSeconds(10));

nikolay-977 commented on src/test/java/test/RegistrationTest.java:
```diff
@@ -0,0 +1,116 @@
+        webDriver.quit();
+    }
+
+//    private void deleteUser() {
```
Необходимо исправить: удалить лишний комментированный код

TimurSharibov committed:
Добавлен родительстий тест. All test runs

TimurSharibov committed:
1.Добавлено неявное ожидание в родительский тест.
2. Удален закомментированный и неиспользуемый код.
3. Добавлена аннотация  @DisplayName в тестах.

TimurSharibov committed:
add allure-report
